services/libs/opensearch/src/service/member.sync.service.ts

```diff
@@ -135,9 +135,7 @@ export class MemberSyncService {
 
       if (toRemove.length > 0) {
         this.log.warn({ tenantId, toRemove }, 'Removing members from index!')
-        for (const id of toRemove) {
-          await this.openSearchService.removeFromIndex(id, OpenSearchIndex.MEMBERS)
-        }
+        await this.openSearchService.bulkRemoveFromIndex(toRemove, OpenSearchIndex.MEMBERS)
```

Comment:
maybe go even further and gather ids in toRemove until you reach the maximum allowed for bulk remove?

Reply:
Good point! There's no limit for bulk removal in OpenSearch; we can do as much as we want, but that also affects the time taken to do the operation on the passed IDs.

I thought this should be safe and quick 🤔

Reply:
What I mean was:
currently we do cleanup by `tenantId` like this:
```
while (nextPage) {
  memberIds = fetchFromOpensearch
  memberIdsNotInDb = filterThroughDb(memberIds)
  delete memberIdsNotInDb from OpenSearch
}  
```

what if we do smt like this:
```
memberIdsNotInDb = []
while (nextPage) {
  memberIds = fetchFromOpensearch
  memberIdsNotInDb.push(filterThroughDb(memberIds))
  if (memberIdsNotInDb.length > SOME_THRESHOLD) {
      delete memberIdsNotInDb from OpenSearch
      memberIdsNotInDb = []
   }
} 

if (memberIdsNotInDb.length > 0) {
 delete memberIdsNoInDb from OpenSearch
} 
```

this is to eliminate too many OpenSearch delete/bulkDelete calls - maybe test it out a bit to see if it pays out on the long run for bigger tenants...

Reply:
Ah gotcha, maybe we can have `SOME_THRESHOLD` as 300.